spacelift/resource_stack.go

```diff
@@ -725,7 +725,12 @@ func uploadStateFile(ctx context.Context, content string, meta interface{}) (str
 		return "", errors.Wrap(err, "could not generate state upload URL")
 	}
 
-	response, err := http.Post(mutation.StateUploadURL.URL, "application/json", strings.NewReader(content))
+	request, err := http.NewRequestWithContext(ctx, http.MethodPut, mutation.StateUploadURL.URL, strings.NewReader(content))
```

Comment:
Probably doesn't matter, but you could consider setting content-type since it was set before. -> `request.Header.Set("Content-Type", "application/json")`

Reply:
Fair point!